Sources/Common/Core/DataArray/index.js

```diff
@@ -313,44 +318,55 @@ function vtkDataArray(publicAPI, model) {
 // Object factory
 // ----------------------------------------------------------------------------
 
-const DEFAULT_VALUES = {
-  name: '',
-  numberOfComponents: 1,
-  size: 0,
-  dataType: DefaultDataType,
-  rangeTuple: [0, 0],
-  // values: null,
-  // ranges: null,
-};
+function defaultValues(initialValues) {
+  return {
+    name: '',
+    numberOfComponents: 1,
+    size: 0,
+    dataType: DefaultDataType,
+    rangeTuple: [0, 0],
+    // values: macro.newTypedArray(DefaultValues),
+    // ranges: null,
+    ...initialValues,
+  };
+}
 
 // ----------------------------------------------------------------------------
 
 export function extend(publicAPI, model, initialValues = {}) {
-  Object.assign(model, DEFAULT_VALUES, initialValues);
-
-  if (!model.empty && !model.values && !model.size) {
+  if (initialValues.empty === false && !initialValues.values) {
```

Comment:
empty flag was intended at construction time so you can allocate the array later on rather than ahead of time. Basically to ensure the user was aware that it created an empty dataArray.

Reply:
looking at [the code in master](https://github.com/Kitware/vtk-js/blob/master/Sources/Common/Core/DataArray/index.js#L329-L341), it seems that an array is ALWAYS provided or created, no ?

Reply:
fair enough, so let's see if we all agree on the following:
```
if initialValues = {empty: true, size: undefined|null|0} then model.values = null
if initialValues = {empty: true, size: 10} then model.values = new TypedArray(10)
if initialValues = {empty: false} then throw error
if initialValues = {empty: false, data: aTypedArray} then model.values = aTypedArray
if initialValues = {empty: false, data: aJSArray} then model.values = TypedArray.from(aJSArray)
if setData(null) then model.values = null
if setData(aTypedArray) then model.values = aTypedArray
if setData(aJSArray) then model.values = TypedArray.from(aJSArray)
```

Reply:
@jourdain Are you ok with https://github.com/Kitware/vtk-js/pull/2409#discussion_r876080336 ?

Reply:
yes, sorry about that

Reply:
For `if initialValues = {empty: true, size: undefined|null|0}`

I will be fine with `model.values = null || TypedArray(0)`

Reply:
> 
> ```
> if initialValues = {empty: true, size: undefined|null|0} then model.values = null
> if initialValues = {empty: true, size: 10} then model.values = new TypedArray(10)
> if initialValues = {empty: undefined|false} then throw error
> if initialValues = {empty: undefined|false, data: aTypedArray} then model.values = aTypedArray
> if initialValues = {empty: undefined|false, data: aJSArray} then model.values = TypedArray.from(aJSArray)
> if setData(null) then model.values = null
> if setData(aTypedArray) then model.values = aTypedArray
> if setData(aJSArray) then model.values = TypedArray.from(aJSArray)
> ```

It should be implemented this way in last commit.